### PR TITLE
Native Binary Build and junit5 prebuilder listener

### PR DESCRIPTION
**Pull request by maschmid**

### CODE REVIEW

File: core/src/main/java/cz/xtf/core/bm/BinaryBuild.java [hunk 1]
```diff
@@ -82,36 +162,69 @@ public void update(OpenShift openShift) {
...
-		openShift.deleteBuildConfig(bc);
+		openShift.imageStreams().withName(is.getMetadata().getName()).delete();
+		openShift.buildConfigs().withName(bc.getMetadata().getName()).delete();
 	}
 
 	@Override
 	public boolean isPresent(OpenShift openShift) {
-		boolean isPresence = openShift.getImageStream(id) != null;
-		boolean bcPresence = openShift.getBuildConfig(id) != null;
+		boolean isPresence = openShift.imageStreams().withName(id).get() != null;
```
Comment (Crumby) [inline core/src/main/java/cz/xtf/core/bm/BinaryBuild.java]: What is the difference?

File: core/src/main/java/cz/xtf/core/bm/BinaryBuild.java [hunk 2]
```diff
@@ -55,23 +126,32 @@ public BinaryBuild(String builderImage, Path path, String contentHash, String id
...
+
+			ExecutorService executorService = Executors.newSingleThreadExecutor();
+			final Future<?> future = executorService.submit(() -> {
+				writeProjectTar(pos);
+			});
+
+			openShift.buildConfigs().withName(bc.getMetadata().getName()).instantiateBinary().fromInputStream(pis);
+			future.get();
+		}
+		catch (IOException e) {
```
Comment (Crumby) [inline core/src/main/java/cz/xtf/core/bm/BinaryBuild.java]: Could we be consistent in formatting? `} catch...`

File: core/src/main/java/cz/xtf/core/bm/BinaryBuild.java [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
+import org.apache.commons.compress.archivers.ArchiveStreamFactory;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.filefilter.TrueFileFilter;
+
 import cz.xtf.core.openshift.OpenShift;
-import cz.xtf.core.openshift.OpenShiftBinary;
-import cz.xtf.core.openshift.OpenShifts;
+
```
Comment (Crumby) [inline core/src/main/java/cz/xtf/core/bm/BinaryBuild.java]: Empty line between several `cz.xtf.core` imports.

File: core/src/main/java/cz/xtf/core/bm/BinaryBuild.java [hunk 4]
```diff
@@ -15,38 +26,98 @@
...
+	private Map<String, String> envProperties;
 
 	private final ImageStream is;
 	private final BuildConfig bc;
 
-	public BinaryBuild(String builderImage, Path path) {
-		this(builderImage, path, ContentHash.get(path));
+	private String contentHash = null;
+
+	private void writeProjectTar(OutputStream os) {
```
Comment (Crumby) [inline core/src/main/java/cz/xtf/core/bm/BinaryBuild.java]: I consider methods ordering constructor, public, private more readable.

File: core/src/main/java/cz/xtf/core/bm/BinaryBuild.java [hunk 5]
```diff
@@ -55,23 +126,32 @@ public BinaryBuild(String builderImage, Path path, String contentHash, String id
...
-
-		openShift.createImageStream(is);
-		openShift.createBuildConfig(bc);
-
-		OpenShiftBinary openShiftBinary = new OpenShiftBinary(OpenShifts.getBinaryPath());
-		if(token != null) {
-			openShiftBinary.login(url, token);
-		} else {
-			openShiftBinary.login(url, username, password);
+
```
Comment (Crumby) [inline core/src/main/java/cz/xtf/core/bm/BinaryBuild.java]: Starting a method with an empty line makes it shattered and harder to read.

File: core/src/main/java/cz/xtf/core/bm/BinaryBuild.java (hunk 1 above)
Comment (maschmid) [reply core/src/main/java/cz/xtf/core/bm/BinaryBuild.java]: I am trying to limit the API use to NamespacedOpenShiftClient whenever it makes sense.

**Pull request merged**